src/main/java/mate/academy/spring/dao/impl/UserDaoImpl.java

```diff
@@ -19,10 +18,9 @@ public UserDaoImpl(SessionFactory factory) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = factory.openSession()) {
-            Query<User> findByEmail = session.createQuery(
-                    "FROM User WHERE email = :email", User.class);
-            findByEmail.setParameter("email", email);
-            return findByEmail.uniqueResultOptional();
+            return session.createQuery(
+                    "FROM User u JOIN FETCH u.roles WHERE email = :email", User.class)
```

Comment:
```suggestion
                    "FROM User u JOIN FETCH u.roles WHERE u.email = :email", User.class)
```
it looks better and will work
p.s. in my implementation code without `u.` doesn't work